core/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -1163,4 +1170,21 @@ public long getLastWriteNanos() {
      * This operation will close the engine if the recovery fails.
      */
     public abstract Engine recoverFromTranslog() throws IOException;
+
+    /**
+     * Called when a refresh includes the location.
+     */
+    public static interface RefreshListener {
+        /**
+         * The location to wait for. Must not return null and should be fast.
+         */
+        Translog.Location location();
```

Comment:
Sorry, I should have said instantiating two classes. No, you are right, we can just use Tuple. I'll do it that way when I pick this up again.